sql/parse/parse.go

```diff
@@ -3381,6 +3381,16 @@ func convertVal(ctx *sql.Context, v *sqlparser.SQLVal) (sql.Expression, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// use the value as string format to keep precision and scale as defined for DECIMAL data type to avoid rounded up float64 value
+		if ps := strings.Split(string(v.Val), "."); len(ps) == 2 {
+			if scale, err := strconv.ParseUint(ps[1], 10, 64); err == nil && scale > 0 {
```

Comment:
Does this work for decimal types with even more precision digits? I'm wondering if this will return an error for a really long decimal component. It would be good to add a test for the max number of decimal digits and make sure we still parse it correctly.

Reply:
Anytime decimal value bigger than specified precision and scale, it creates warning and uses rounded value.